packages/adblocker/src/engine/bucket/preprocessor.ts

```diff
@@ -0,0 +1,133 @@
+import { StaticDataView } from '../../data-view';
+import IFilter from '../../filters/interface';
+import Preprocessor, { Env } from '../../preprocessor';
+
+export default class PreprocessorBucket {
+  public static deserialize(view: StaticDataView): PreprocessorBucket {
+    const preprocessors: Preprocessor[] = [];
+
+    for (let i = 0, l = view.getUint32(); i < l; i++) {
+      preprocessors.push(Preprocessor.deserialize(view));
+    }
+
+    return new this({
+      preprocessors,
+    });
+  }
+
+  public readonly conditions: Map<string, Preprocessor>;
+  public readonly excluded: Set<number>;
+
+  public env: Env;
```

Comment:
We do need to know the `env` only if we need to `update`. However, I don't know if we want to store `env` outside and provide it to `updateFromDiff` function because I don't expect there'll be a direct use of `PreprocessorBucket.update`.

Reply:
So let's say the update method (e.g. `updateFromDiff`) would take `env` as argument, then it would remove the need to store it internally right? I think it's worth considering since it would simplify the internal logic further (less state to keep, less serializing/deserializing logic, etc.)